src/NuGet.Core/NuGet.Protocol/EnhancedHttpRetryHelpers.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    internal class EnhancedHttpRetryHelper
```

Comment:
I would make this class static so each env var is only read once like it was before.  What do you think?

Reply:
> I would make this class static so each env var is only read once like it was before. What do you think?

I initially tried that but ended up in the same boat with needing to feed in environment variable readers from non-static classes, for testing.  Reading environment variables is actually very cheap computationally... I think reading them multiple times isn't the worst thing.

Reply:
I verified that providers like `ServiceIndexResourceV3Provider` are only instantiated once so at least the env vars won't be read N-times per package.  So this should be fine unless anyone else feels otherwise.

Reply:
Yeah, we do plenty of caching as long as the same `SourceRepository` is used.

Reply:
https://github.com/NuGet/NuGet.Client/pull/4238#pullrequestreview-745387230

Reply:
https://github.com/NuGet/NuGet.Client/pull/4238#pullrequestreview-745387230

Reply:
https://github.com/NuGet/NuGet.Client/pull/4238#pullrequestreview-745387230

Reply:
https://github.com/NuGet/NuGet.Client/pull/4238#pullrequestreview-745387230